### DISCUSSION TITLE
Set Administration Mode via PAC CLI

### DISCUSSION THREAD

**Discussion by jboliveira** [microsoft/powerplatform-build-tools#475] [Ideas]
### Idea
Currently, in order to set administrative mode and/or background operations due to significant operational changes, we must do the following steps: https://learn.microsoft.com/en-us/power-platform/admin/admin-mode#set-administration-mode

As an alternative, the following PowerShell script may be used to perform such an operation: _[version isn't complete]_

```powershell
$token = "***"
$uri = "https://api.bap.microsoft.com/providers/Microsoft.BusinessAppPlatform/environments/$($EnvironmentId)?api-version=2020-06-01"
$headers = @{
        "Accept"        = "application/json, text/plain, */*"
        "authorization" = "Bearer $token"
}

# Enable admin mode and disable background operations
$payload = "{`"properties`":{`"states`":{`"runtime`":{`"id`":`"AdminMode`"}},`"linkedEnvironmentMetadata`":{`"backgroundOperationsState`":`"Disabled`"}}}"

Invoke-RestCall -u $uri -m Patch -h $headers -b $payload
```

However, it would be excellent if such a functionality could be accessed using the PAC CLI (`pac admin`).

### Commands

Usage: `pac admin mode [--enableAdmin] [--disableBackgroundOperations] [--status]`

`--enableAdmin` - Choose to enable the Administration Mode. The default value is 'true'. _(alias: -a)_
`--disableBackgroundOperations` - Choose to disable Background Operations when enabling the Admin mode. The default value is 'true'. _(alias: -bo)_
`--status` - Show the current status of Administration Mode and Background Operations. _(alias: -s)_

Examples:

```powershell
# Enable Admin mode and keep Background Operations enabled.
pac admin mode --enableAdmin

# Enable Admin mode and disable Background Operations.
pac admin mode --enableAdmin --disableBackgroundOperations

# Disable Admin mode and enable Background Operations implicitly.
pac admin mode --enableAdmin false

# Show current status of Admin mode and Background Operations.
pac admin mode --status
```

**Comment by marcelbf** [comment]
Thanks for the suggestion. I've raised this request to the proper team. It helps if you also post this on https://ideas.powerapps.com/, under category governance. Thanks!

**Reply by jboliveira** [reply]
Thank you for adding the suggestion to the CLI.
